indico/modules/events/contributions/forms.py

```diff
@@ -20,20 +20,25 @@
 
 from wtforms.ext.sqlalchemy.fields import QuerySelectField
 from wtforms.fields import StringField, TextAreaField
-from wtforms.validators import DataRequired
+from wtforms.validators import DataRequired, ValidationError
 
 from indico.modules.events.contributions.fields import ContributionPersonListField, SubContributionPersonListField
 from indico.modules.events.contributions.models.references import ContributionReference, SubContributionReference
 from indico.modules.events.fields import ReferencesField
 from indico.web.forms.base import IndicoForm
-from indico.web.forms.fields import TimeDeltaField, PrincipalListField, IndicoLocationField, IndicoProtectionField
-from indico.web.forms.validators import UsedIf, MaxDuration
+from indico.web.forms.fields import (TimeDeltaField, PrincipalListField, IndicoProtectionField, IndicoLocationField,
+                                     IndicoDateTimeField)
+from indico.web.forms.validators import UsedIf, DateTimeRange, MaxDuration
 from indico.util.i18n import _
 
 
 class ContributionForm(IndicoForm):
     title = StringField(_("Title"), [DataRequired()])
     description = TextAreaField(_("Description"))
+    start_date = IndicoDateTimeField(_("Start date"), [DataRequired(),
+                                                       DateTimeRange(earliest=lambda f, field: f.conf.getStartDate(),
```

Comment:
Since we have the start/end date in Event now you can use `f.event.start_dt`/`f.event.end_dt` and possibly get rid of `self.conf = ..`